### PR TITLE
LPS-152081 Convert MailService to MainSender and move to apps/portal/portal-mail

### PR DESCRIPTION
**Pull request by julschong**
Hi Tina,

This is part 1 of removing petra-mail: [LPS-148669](https://issues.liferay.com/browse/LPS-148669)
Subtask: [LPS-152081 Convert MailService to MainSender and move to apps/portal/portal-mail](https://issues.liferay.com/browse/LPS-152081)

previously tested on my repo here: https://github.com/julschong/liferay-portal/pull/336#issuecomment-1124418608

Thank you!

### CODE REVIEW

File: modules/apps/portal/portal-mail/src/main/java/com/liferay/portal/mail/sender/internal/messaging/MailSenderMessagingConfigurator.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+		activate(bundleContext, properties);
+	}
+
+	private volatile BundleContext _bundleContext;
+
+	@Reference
+	private DestinationFactory _destinationFactory;
+
+	private volatile MailSenderMessagingConfiguration
```
Comment (dantewang) [inline modules/apps/portal/portal-mail/src/main/java/com/liferay/portal/mail/sender/internal/messaging/MailSenderMessagingConfigurator.java:105]: @julschong This field is only used in `activate` method, please make it a method variable.

File: modules/apps/portal/portal-mail/src/main/java/com/liferay/portal/mail/sender/internal/messaging/MailSenderMessageListener.java [hunk 2]
```diff
@@ -171,10 +119,75 @@ protected InternetAddress[] filterInternetAddresses(
...
+
 	private static final Log _log = LogFactoryUtil.getLog(
-		MailMessageListener.class);
+		MailSenderMessageListener.class);
 
 	private static final Set<String> _mailSendBlacklist = new HashSet<>(
 		Arrays.asList(PropsValues.MAIL_SEND_BLACKLIST));
 
+	@Reference
+	private MailSender _mailSender;
```
Comment (dantewang) [inline modules/apps/portal/portal-mail/src/main/java/com/liferay/portal/mail/sender/internal/messaging/MailSenderMessageListener.java:191]: @julschong This reference is not used. Just to make sure, did you add it to ensure the listener starts after the `MailSenderImpl` OSGi service?

**Review by dantewang — COMMENTED** [review]
@julschong Please see my comments. cc @tinatian

**Comment by julschong** [comment]
@tinatian @dantewang
I will resend to preserve Dante's comments

File: modules/apps/portal/portal-mail/src/main/java/com/liferay/portal/mail/sender/internal/MailSenderImpl.java [hunk 3]
```diff
@@ -68,7 +71,9 @@ public void addForward(
 			_addForwardMethodKey, companyId, userId, filters, emailAddresses,
 			leaveCopy);
 
-		MessageBusUtil.sendMessage(DestinationNames.MAIL, methodHandler);
+		MessageBusUtil.sendMessage(
+			MailSenderMessagesDestinationNames.MAIL_SENDER_MESSAGES_PROCESSOR,
```
[the remark this answers is not in the document]
Comment (julschong) [reply modules/apps/portal/portal-mail/src/main/java/com/liferay/portal/mail/sender/internal/MailSenderImpl.java:75]: I will remove `MailSenderMessageDestinationNames`, I was focused on using MessageBus as a reference for making my change. so `MailSenderMessageDestinationNames` was not needed

File: modules/apps/portal/portal-mail/src/main/java/com/liferay/portal/mail/sender/internal/MailSenderImpl.java [hunk 4]
```diff
@@ -34,6 +32,7 @@
 import com.liferay.portal.kernel.util.PropertiesUtil;
 import com.liferay.portal.kernel.util.PropsKeys;
 import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.mail.sender.internal.constants.MailSenderMessagesDestinationNames;
```
[the remark this answers is not in the document]
Comment (julschong) [reply modules/apps/portal/portal-mail/src/main/java/com/liferay/portal/mail/sender/internal/MailSenderImpl.java:35]: will remove `MailSenderMessageDestinationNames`

File: modules/apps/portal/portal-mail/src/main/java/com/liferay/portal/mail/sender/internal/messaging/MailSenderMessageListener.java (hunk 2 above)
Comment (julschong) [reply modules/apps/portal/portal-mail/src/main/java/com/liferay/portal/mail/sender/internal/messaging/MailSenderMessageListener.java:191]: will do. No, I'm pretty sure I did thing by mistake

**Pull request closed without merging**